manual/trees/index.md

```diff
@@ -7,178 +7,145 @@ toc: true
+If told, TTree will create (sub-) branches for each member of a class and its base classes.
+If such a member is a class itself, that member's type can also be split.
+The recusion level of nested splitting is called the "split level"; it can be configured during branch creation.
 
-{% highlight C++ %}
-root [0] .x cernbuild.C
-******************************************************************************
-*Tree    :T         : CERN 1988 staff data                                   *
-*Entries :     3354 : Total =          176339 bytes  File  Size =      15005 *
-*        :          : Tree compression factor =   2.74                       *
-******************************************************************************
-*Br    0 :Category  : Category/I                                             *
-*Entries :     3354 : Total  Size=      14073 bytes  One basket in memory    *
-*Baskets :        0 : Basket Size=      32000 bytes  Compression=   1.00     *
-*............................................................................*
-*Br    1 :Flag      : Flag/i                                                 *
-*Entries :     3354 : Total  Size=      14049 bytes  One basket in memory    *
-*Baskets :        0 : Basket Size=      32000 bytes  Compression=   1.00     *
-*............................................................................*
-*Br    2 :Age       : Age/I                                                  *
-*Entries :     3354 : Total  Size=      14043 bytes  One basket in memory    *
-*Baskets :        0 : Basket Size=      32000 bytes  Compression=   1.00     *
-*............................................................................*
-*Br    3 :Service   : Service/I                                              *
-*Entries :     3354 : Total  Size=      14067 bytes  One basket in memory    *
-*Baskets :        0 : Basket Size=      32000 bytes  Compression=   1.00     *
-*............................................................................*
-...
-...
-...
-{% endhighlight %}
+If the split level is set to 0, there is no splitting: all data members are stored in the same branch.
+Data members can also be configured to be non-split as part of the dictionary; see → [I/O]({{ '/manual/io' | relative_url }}).
+The default split level is 1; a split level of 99 means to split all members at any recursion level.
 
-### Creating a tree from a folder structure
+_Pointers_
 
-You can build a folder structure and create a tree with branches for each of the sub-folders.
+While references `X &` are not supported as member types, pointers are.
+If the pointer is non-null, ROOT stores the object pointed to (pointee).
+If multiple pointers point to the same object during one `TTree::Fill()` operation, that pointee will only be stored once; upon reading, all pointers will again point to the same object.
```

Comment:
Technically that is during on `TBranch::Fill`.  For example if we have (mostly likely indirectly rather than this improbable simplification :) ):
```
struct SomeClass {
   Other *fPointer;
   Ohter *fAnotherPointer; 
};
```
and `fPointer` and `fAnotherPointer` points to the same address **and** the struct `SomeClass` is stored with split level 1 or higher, then the pointee object will be stored twice (and more annoyingly when retrieving the object, `fPointer` and `fAnotherPointer` will not point to two distinct objects .... hence the need/recommendation to use references (like TRef) or indices instead of pointers.

Reply:
Thanks for the clarification! Addressed.